smdebug/tensorflow/keras.py

```diff
@@ -1351,31 +1372,43 @@ def profiling_start_batch(self, mode=ModeKeys.TRAIN):
 
         self.profiler_config_parser.load_config()
         self.profiler_config_parser.handle_step_start_python_profiling(mode, current_step)
+        self._start_or_stop_dataloader_profiling(current_step)
 
     def profiling_end_batch(self, mode=ModeKeys.TRAIN):
         """
         Enabling profiler at the end of train batch for native Tf2 training.
         :param mode: ModeKeys.TRAIN ModeKeys.EVAL ModeKeys.PREDICT
 
-        TODO: Add support for detailed, dataloader and SMDDP profiling at the end of the batch.
+        TODO: Add support for detailed and SMDDP profiling at the end of the batch.
         """
         if self._is_not_supported():
             return
 
+        self.record_trace_events(
+            training_phase="Step:" + str(mode),
+            op_name="Step:" + str(mode),
+            phase="X",
```

Comment:
Can we manage phase names with Enums?

Reply:
That'll require refactoring throughout the codebase and should be out of scope for this PR.